### PR TITLE
Client: Option to change replica sort algorithm in download client Closes #5355

### LINKED ISSUE #5355: Option to change replica sort algorithm in download client
Motivation
----------

The current implementation of the rucio download client will call `list_replicas` API to get a sorted replica list, and then try to download data from replicas in order. But the download client doesn't pass any sorting algorithm selection to the `list_replicas` API. Since `list_replicas` has already supported such an option, can we also let the download client support the same option? For example,

```
$ rucio download --replica-selection SORT ...
```

Modification
------------

Add a new option `--replica-selection` to the rucio download cli, and a new argument `sort` to `download_dids` API.

### PR DESCRIPTION
**Pull request by fno2010**
Allow the rucio download client to specify the replica sorting algorithm using CLI as follows:

```
rucio download --replica-selection random test:file1
```

or using the client API as follows:

```
download_client.download_dids(items, sort="random")
```

### CODE REVIEW

File: bin/rucio [hunk 1]
```diff
@@ -2390,6 +2393,7 @@ You can filter by key/value, e.g.::
         selected_parser.add_argument('--scope', dest='scope', action='store', help='Scope if you are using the filter option and no full DID.')
         selected_parser.add_argument('--metalink', dest='metalink_file', action='store', help='Path to a metalink file.')
         selected_parser.add_argument('--deactivate-file-download-exceptions', dest='deactivate_file_download_exceptions', action='store_true', help='Does not raise NoFilesDownloaded, NotAllFilesDownloaded or incorrect number of output queue files Exception.')  # NOQA: E501
+        selected_parser.add_argument('--replica-selection', dest='sort', action='store', help='Select the best replica using a replica sorting algorithm.')
```
Comment (mlassnig) [inline bin/rucio]: give a list of possible values in the comment

**Review by mlassnig — COMMENTED** [review]
LGTM, just one tiny UX comment

**Comment by bari12** [comment]
Looks good to me too. Can you please rebase this, there was a quite big PR these days which changed most of the files, that's why this probably now has merge conflicts.

**Comment by bari12** [comment]
Looks good to me too. Can you please rebase this, there was a quite big PR these days which changed most of the files, that's why this probably now has merge conflicts.

**Comment by fno2010** [comment]
> Looks good to me too. Can you please rebase this, there was a quite big PR these days which changed most of the files, that's why this probably now has merge conflicts.

I have rebased this PR two weeks ago. But it seems that there have been several commits again after the rebase point. Although I didn't see any further conflicts, shall I rebase it again?

**Review by bari12 — APPROVED** [review]

**Pull request merged**